### PR TITLE
Adding UIDeploymentRunner class and example variables.yaml

### PR DESCRIPTION
**Pull request by blrm**
This adds a deployment runner class that reads in a dictionary from a variables.yaml file (example file included) and supplies a set of methods to interact with corresponding page objects using the values from the supplied yaml file. Each method takes a page object as a parameter, fills out the page, and returns the next page object from the wizard. 

This should let us write deployment test cases that can determine a path through the wizard based on the page object class which is returned instead of checking the yaml data itself. For instance, if you are clicking next on the Access Insights page and wanted to check if you are on RHEV page:

```python
runner = UIDeploymentRunner()                                   
deployment_name_pg = runner.product_selection(new_deployment_pg)
update_avail_pg = runner.deployment_name(deployment_name_pg)    
insights_pg = runner.update_availability(update_avail_pg)       
next_pg = runner.access_insights(insights_pg)                   
# check if we are on the RHV Setup Type page                    
if isinstance(next_pg, SetupType):                              
    setuptype_pg = next_pg                                      
```

### CODE REVIEW

**Comment by apagac** [comment]
:+1:

File: lib/deployment_runner.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
+import yaml
```
Comment (jamesooden) [inline lib/deployment_runner.py]: I would put a header comment here that describes the purpose of this library, and how it basically works.

**Comment by jamesooden** [comment]
:+1:

**Comment by jamesooden** [comment]
Here are some thoughts that I don't expect action right yet, but I just wanted to put them out there.

- What could we do to separate the actual config file format from the runner?   It seems the runner should not have any knowledge of the config file format, and should just have methods to input the data for each type of deployment.
- Would it be helpful to split each type of deployment into a separate module?   It seems right now the runner has far too much information about each type of deployment.   Maybe we could have a directory where all the deployment types are loaded up automatically from.   Think of it as a product level driver.
- Would it be possible once all the information is given to the runner for it to just run a deployment?   Why even have the logic in a deployment test to figure out what was the next module/page?   Obviously being able to stop at a certain point so a given test can intervene would be good, but for just running through a particular type of deployment, couldn't that be completely automatic?

**Comment by jamesooden** [comment]
:+1: 

Those latest changes look good.

**Comment by apagac** [comment]
:+1: 
Looks good to me

**Pull request merged**